grype/matcher/matchers.go

```diff
@@ -81,9 +61,26 @@ func (c *controller) trackMatcher() (*progress.Manual, *progress.Manual) {
 	return &packagesProcessed, &vulnerabilitiesDiscovered
 }
 
-func (c *controller) findMatches(provider vulnerability.Provider, release *linux.Release, packages ...pkg.Package) match.Matches {
+func newMatcherIndex(matchers []Matcher) map[syftPkg.Type][]Matcher {
```

Comment:
I wonder if it would be better to do something like:

```
type Matchers map[syftPkg.Type][]Matcher
```
and then instead of having this function, just return the `Matchers` type like `func NewDefaultMatchers(mc Config) Matchers {` above?

Reply:
From a library standpoint, I think it's good to have the list returned. We've chosen to use this list to build the `syftPkg.Type` map for our own lookup purposes, but we should not exclude users from being able to also run `NewDefaultMatchers()` and use them in different ways
```
for _, m := range matchers {
  m.SomeFunctionality
}
```

Reply:
Also, I don't know how I feel creating an exported custom type that is a coupling of the two tools together in this case.

Reply:
@wagoodman @luhring @jonasagx thoughts here?

Reply:
Agree fine as is and it does become a little odd to export something based on the syft data model, given the current way the grype data model works.